src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -58,6 +58,22 @@ public static LibraryVersion GetInstalledAsmVersion(List<LibraryVersion> version
                         return (LibraryVersion)install.Value.Item1;
                     }
                 }
+
+                //Fallback mechanism, look inside libg folders if any of them
+                //contain ASM dlls.
+                var rootDir = new DirectoryInfo(rootFolder);
+                var dirs = rootDir.GetDirectories("libg_*");
+                foreach (var dir in dirs)
+                {
+                    var files = dir.GetFiles("ASMAHL*.dll");
+                    if (files.Any())
+                    {
+                        location = dir.FullName;
+                        return
+                            versions.FirstOrDefault(
+                                v => dir.Name.Equals(string.Format("libg_{0}", (int)v)));
+                    }
+                }                
```

Comment:
This looks reasonable. The only thing is it does all heavy lifting upfront -- enumerating directory names and then get files with `ASMAHL` prefix, then eventually try to see if `versions` list contain the requested entry. Instead, what could have been slightly better is to base the folder name on the entries in `versions` list, then do the file retrieval:

```csharp
foreach (var v in versions)
{
    var folderName = string.Format("libg_{0}", v);
    if (!Directory.Exists(folderName))
        continue;

    // Pseudo code, such method doesn't exist.
    if (!Directory.ContainsFile("ASMAHL*.dll"))
        continue;

    return v; // Found version.
}
```